### PR TITLE
successful offer creation returns status code 201

### LINKED ISSUE #4: create offer should return status code 201 instead of 422 for success
At the moment the response code is 422 whether the offer was created or not.

### PR DESCRIPTION
**Pull request by kat-ka**
As HashMap put method is defined, the return value for creating an Offer in SimpleInMemoryRepository was empty.

Use response code `201 'Created'` instead of `202 'Accepted'` for Offer created successfully.

Use response code `400/415` for wrong data input/wrong format.

Use thread save HashMap.

Tests are included.

### CODE REVIEW

File: src/main/java/org/sharesquare/hub/endpoints/Offers.java [hunk 1]
```diff
@@ -57,16 +64,24 @@
     }
 
     @Operation(description = "Create an Offer")
-    @ApiResponse(description = "Successful operation", responseCode = "202")
-    @ApiResponse(description = "Malformed Data", responseCode = "422", content = @Content)
+    @ApiResponse(responseCode = "201", description = "Success")
+    @ApiResponse(responseCode = "415", description = "Wrong format/data", content = @Content)
```
Comment (dividebyzero) [inline src/main/java/org/sharesquare/hub/endpoints/Offers.java]: IMHO it is wrong to just replace 422 with 415 in all cases. 415 should be used if the data type of the request is wrong, 422 if the data itself has errors.

>  The 422 (Unprocessable Entity) status code means the server
   understands the content type of the request entity (hence a
   415(Unsupported Media Type) status code is inappropriate), and the
   syntax of the request entity is correct (thus a 400 (Bad Request)
   status code is inappropriate) but was unable to process the contained
   instructions.  For example, this error condition may occur if an XML
   request body contains well-formed (i.e., syntactically correct), but
   semantically erroneous, XML instructions.

> 

(https://tools.ietf.org/html/rfc4918#section-11.2)

Comment (kat-ka) [reply src/main/java/org/sharesquare/hub/endpoints/Offers.java]: Makes sense. Not sure, but maybe also use 400:
https://tools.ietf.org/html/rfc7231
like mentioned in section _6.5.1. 400 Bad Request_

422 was recommended till 2014

Comment (kat-ka) [reply src/main/java/org/sharesquare/hub/endpoints/Offers.java]: Kind of read that the use of 422 in this case is rather outdated and 400 should be used instead. Acceptable?

Comment (kat-ka) [reply src/main/java/org/sharesquare/hub/endpoints/Offers.java]: Changed it in the code.

**Review by dividebyzero — APPROVED** [review]

**Pull request merged**